ui/raidboss/data/05-shb/raid/e8s.js

```diff
@@ -317,30 +317,201 @@
+              )
+            );
+            if ((tankTether1 === 'NE' && tankTether2 === 'SW') ||
+                (tankTether1 === 'SW' && tankTether2 === 'NE')) {
+              // If the tank is tethered to NE and SW (makes a box)
+              // Tank swaps with SW to make bowtie
+              data.rampant.chain.chainSwapTank = 'SW';
+              data.rampant.chain.chainSwapDps = data.rampant.chain.chains.dps1.source;
+            } else if ((tankTether1 === 'NE' && tankTether2 === 'SE') ||
+                (tankTether1 === 'SE' && tankTether2 === 'NE')) {
+              // If the tank is tethered to NE and SE (makes an hourglass)
+              // Tank swaps with SE to make bowtie
+              data.rampant.chain.chainSwapTank = 'SE';
+              data.rampant.chain.chainSwapDps = data.rampant.chain.chains.dps2.source;
+            }
+          }
+          return data.rampant.chain.total===4;
+        }
+      },
+      infoText: function(data, matches) {
+        if (data.options.cactbote8sLightRampantStrat === undefined || data.options.cactbote8sLightRampantStrat === 'none') {
+          return {
+            en: 'Chain on YOU',
+            de: 'Kette auf DIR',
+            fr: 'Chaine sur VOUS',
+            ko: '사슬 대상자',
+            cn: '连线',
+          };
+        }
+
+        if (data.options.cactbote8sLightRampantStrat === 'sharingan') {
+          let ret = {};
+          if (data.rampant.chain.meChain) {
+            let extra = '';
+            if (data.rampant.chainSwapTank!=='Stay') {
+              if (data.rampant.chain.chains.tank.source == data.me)
+                extra = ', Swap '+data.rampant.chain.chainSwapTank;
+              else if (data.rampant.chainSwapDps == data.me)
+                extra = ', Swap NW';
+              else if (data.party.isHealer(data.me))
+                extra = ', Tank swap with '+data.rampant.chain.chainSwapTank;
+            }
+            ret = {
+              en: 'Chain on YOU'+extra,
+              de: 'Kette auf DIR',
+              fr: 'Chaine sur VOUS',
+              ko: '사슬 대상자',
+              cn: '连线',
+            };
+          }
+          delete data.rampant;
+          return ret;
+        }
       },
     },
     {
       id: 'E8S Holy Light',
       regex: Regexes.tether({ id: '0002' }),
       condition: Conditions.targetIsYou(),
-      infoText: {
-        en: 'Orb on YOU',
-        de: 'Orb auf DIR',
-        fr: 'Orbe sur VOUS',
-        ko: '구슬 대상자',
-        cn: '拉球',
+      promise: function(data, matches) {
+        if (data.options.cactbote8sLightRampantStrat === undefined || data.options.cactbote8sLightRampantStrat === 'none')
+          return null;
+
+        let p = new Promise(async (res) => {
+          let combatantData = await window.callOverlayHandler({
+            call: 'getCombatants',
+            ids: [matches.sourceId],
+          });
+          data.rampant = data.rampant || {};
+          if (!(combatantData !== null &&
```

Comment:
Can you apply this "not" through and make it combatantData === null || !combatantData.combatants || !combatantData.combatants.length to make this easier to read?

Reply:
Adjusted in local git.